cmd/lakectl/cmd/local_checkout.go

```diff
@@ -15,73 +18,94 @@ var localCheckoutCmd = &cobra.Command{
 	Short: "Sync local directory with the remote state.",
 	Args:  localDefaultArgsRange,
 	Run: func(cmd *cobra.Command, args []string) {
-		_, localPath := getLocalArgs(args, false)
 		syncFlags := getLocalSyncFlags(cmd)
 		specifiedRef := Must(cmd.Flags().GetString("ref"))
-		idx, err := local.ReadIndex(localPath)
-		if err != nil {
-			if errors.Is(err, fs.ErrNotExist) {
-				DieFmt("directory %s is not linked to a lakeFS path", localPath)
-			}
-			DieErr(err)
+		all := Must(cmd.Flags().GetBool("all"))
```

Comment:
--all and --ref are mutually exclusive. ref is relevant only in the context of a single directory